src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1465,6 +1465,11 @@ private static bool NormalizeTaskTypesInArray(CSharpCompilation compilation, ref
         private static bool NormalizeTaskTypesInPointer(CSharpCompilation compilation, ref PointerTypeSymbol pointerType)
         {
             var pointedAtType = pointerType.PointedAtType;
+            if (pointedAtType.SpecialType == SpecialType.System_Void)
+            {
+                // IsCustomTaskType() is not expecting System.Void.
```

Comment:
>// IsCustomTaskType() is not expecting System.Void. [](start = 16, length = 51)

Would it be better to adjust IsCustomTaskType implementation instead? Otherwise, each caller (there are several now) has to have a special case.